src/Api/UserApi.php

```diff
@@ -12,6 +17,31 @@ class UserApi implements UserApiInterface
    */
   private $token;
 
+  /**
+   * @var ValidatorInterface
+   */
+  private $validator;
+
+  /**
+   * @var UserManager
+   */
+  private $user_manager;
+
+  /**
+   * @var TokenGenerator
+   */
+  private $token_generator;
+
+  /**
+   * UserApi constructor.
+   */
+  public function __construct(ValidatorInterface $validator, UserManager $user_manager, TokenGenerator $token_generator)
+  {
+    $this->validator = $validator;
+    $this->user_manager = $user_manager;
+    $this->token_generator = $token_generator;
```

Comment:
Regarding our discussion yesterday about the FOSUserBundle `user_manager` already extends the correct Controller so no need to do anything here :)